File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/tasks/DriveTrainTask.java
package org.firstinspires.ftc.teamcode.tasks;

import com.qualcomm.hardware.bosch.BNO055IMU;
import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
import com.qualcomm.robotcore.hardware.DcMotor;
import com.qualcomm.robotcore.hardware.DcMotorSimple;
import com.qualcomm.robotcore.util.ElapsedTime;

import org.firstinspires.ftc.teamcode.robotutil.DriveTrainNew;
import org.firstinspires.ftc.teamcode.robotutil.Utils;
import org.firstinspires.ftc.teamcode.robotutil.IMU;

/**
 * Created by Howard on 10/15/16.
 */
public class DriveTrainTask extends TaskThread {
    DriveTrainNew dt;

    public DriveTrainTask(LinearOpMode opMode) {
        this.opMode = opMode;

        initialize();
    }

    @Override
    public void run() {
        timer.reset();
        while (opMode.opModeIsActive() && running) {

            dt.setPowers(opMode.gamepad1.left_stick_y,opMode.gamepad1.right_stick_y);
        }
        dt.setPowers(0,0);
    }
    @Override
    public void initialize() {

        this.dt = new DriveTrainNew(this.opMode);
        dt.initMotors();
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Tasks/DriveTrainTask.kt
package org.firstinspires.ftc.teamcode.Tasks

import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode
import org.firstinspires.ftc.teamcode.Components.DriveTrain
import org.firstinspires.ftc.teamcode.Models.Task

public class DriveTrainTask(opMode:LinearOpMode, taskName:String): Task(opMode,taskName){
    var dt = DriveTrain(opMode)

    override fun run(){
        runWhileActive(this::main)
    }

    fun main(){
        dt.setPowers(
                opMode.gamepad1.left_stick_y.toDouble(),
                opMode.gamepad1.right_stick_y.toDouble()
        );    }

}